src/common/signalk/SKNMEAVisitor.cpp

```diff
@@ -71,9 +73,77 @@ void SKNMEAVisitor::visitSKNavigationAttitude(const SKUpdate &u, const SKPath &p
 };
 
 void SKNMEAVisitor::visitSKNavigationHeadingMagnetic(const SKUpdate &u, const SKPath &p, const SKValue &v) {
-  NMEASentenceBuilder sb2("II", "HDM", 2);
-  sb2.setField(1, SKRadToDeg(v.getNumberValue()), 1);
-  sb2.setField(2, "M");
+  NMEASentenceBuilder sb("II", "HDM", 2);
+  sb.setField(1, SKRadToDeg(v.getNumberValue()), 1);
+  sb.setField(2, "M");
 
-  _sentences.add(sb2.toNMEA() + "\r\n");
+  _sentences.add(sb.toNMEA() + "\r\n");
 };
+
+// ***********************  Wind Speed and Angle  ************************
+//  Talker ID: WI Weather Instruments
+//             VW Velocity Sensor, Mechanical?
+//  also seen: IIVWR
+//
+//  NMEA0183  MWV Wind Speed and Angle  --> recommended
+//              1  2  3  4
+//              |  |  |  |
+//      $--MWV,x.x,a,x.x,a*hh
+//  1) Wind Angle, 0 to 360 degrees
+//  2) Reference, R = Relative, T = True
+//  3) Wind Speed
+//  4) Wind Speed Units: K = km/s, M = m/s, N = Knots
+//  5) Status, A = Data Valid
+// ************************************************************************
+void SKNMEAVisitor::visitSKEnvironmentWindAngleApparent(const SKUpdate &u, const SKPath &p, const SKValue &v) {
+
+  float windAngle;
+  float windSpeed;
+  windAngle = v.getNumberValue();
+
+  // Look if we have an update for speed too
+  if (!u.hasEnvironmentWindSpeedApparent()) return;
+
+  windSpeed = u.getEnvironmentWindSpeedTrue();  // in m/s
+
+  NMEASentenceBuilder sb( "II", "MWV", 5);
+  sb.setField(1, SKRadToDeg(windAngle), 2 );
```

Comment:
I just realized that when we parse the angle from NMEA2000 we do not apply `SKNormalizeDirection()` so it's possible that the angle is negative (-35 degrees for example). However `MWV` specifically calls for an angle between 0 and 360 so I think we should normalize to a direction here. Do you agree?

Reply:
Yes, true! Added and pushed.

In the HDM (Heading Magnetic) should there be a NormalizeDirection too?

And in the NMEA2000Parser itself I miss your nice macro?
```
case N2kWind_True_North:
  // Ground Wind Direction
  update->setEnvironmentWindDirectionTrue(windAngle); --> SKNormalizeDirection()

case N2kWind_Magnetic:
  // Ground Wind Direction referred to magnetic north
  update->setEnvironmentWindDirectionMagnetic(windAngle); --> SKNormalizeDirection()
    
```